src/d/actor/d_a_obj_eff.cpp

```diff
@@ -2,261 +2,479 @@
+        return true;
+    }
+
+    return false;
 }
 
 /* 00000E80-00000F6C       .text eff_set__Q28daObjEff5Act_cFv */
-void daObjEff::Act_c::eff_set() {
-    /* Nonmatching */
+BOOL daObjEff::Act_c::eff_set() {
+    static Proc proc[] = {
+        &eff_set_barrel_smoke,
+        &eff_set_stool_smoke,
+        &eff_set_skull_smoke,
+        &eff_set_land_smoke,
+        &eff_set_pinecone_smoke,
+        &eff_set_woodBox_smoke
+    };
+
+    return (this->*proc[mProcIndex])();
 }
 
 /* 00000F6C-00001030       .text _create__Q28daObjEff5Act_cFv */
 cPhs_State daObjEff::Act_c::_create() {
-    /* Nonmatching */
+    static const u32 heap_size[] = {
+        0x00000040,
+        0x00000040,
+        0x00000040,
+        0x00000020,
+        0x00000040,
+        0x00000040,
+    };
+
+    mProcIndex = daObj::PrmAbstract(this, PRM_ARG_W, PRM_ARG_S);
+    fopAcM_SetupActor(this, daObjEff::Act_c);
+    int phase = cPhs_COMPLEATE_e;
+    if (fopAcM_entrySolidHeap(this, &solidHeapCB, heap_size[mProcIndex])) {
+        if ((u8) eff_set() != 0) {
+            fopDwTg_DrawQTo(&draw_tag);
+        } else {
+            phase = cPhs_ERROR_e;
+        }
+    } else {
+        phase = cPhs_ERROR_e;
+    }
+
+    return phase;
 }
 
 /* 00001030-00001068       .text remove_barrel_smoke__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove_barrel_smoke() {
-    /* Nonmatching */
+    if (mParticleCallback != NULL) {
+        static_cast<BarrelSmokeCB *>(mParticleCallback)->end();
+    }
 }
 
 /* 00001068-000010A0       .text remove_stool_smoke__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove_stool_smoke() {
-    /* Nonmatching */
+    if (mParticleCallback != NULL) {
+        static_cast<StoolSmokeCB *>(mParticleCallback)->end();
+    }
 }
 
 /* 000010A0-000010D8       .text remove_skull_smoke__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove_skull_smoke() {
-    /* Nonmatching */
+    if (mParticleCallback != NULL) {
+        static_cast<SkullSmokeCB *>(mParticleCallback)->end();
+    }
 }
 
 /* 000010D8-00001110       .text remove_land_smoke__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove_land_smoke() {
-    /* Nonmatching */
+    if (mParticleCallback != NULL) {
+        static_cast<LandSmokeCB *>(mParticleCallback)->end();
+    }
 }
 
 /* 00001110-00001148       .text remove_pinecone_smoke__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove_pinecone_smoke() {
-    /* Nonmatching */
+    if (mParticleCallback != NULL) {
+        static_cast<PineconeSmokeCB *>(mParticleCallback)->end();
+    }
 }
 
 /* 00001148-00001180       .text remove_woodBox_smoke__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove_woodBox_smoke() {
-    /* Nonmatching */
+    if (mParticleCallback != NULL) {
+        static_cast<WoodBoxSmokeCB *>(mParticleCallback)->end();
+    }
 }
 
 /* 00001180-0000126C       .text remove__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove() {
-    /* Nonmatching */
+    static voidProc proc[] = {
+        &remove_barrel_smoke,
```

Comment:
All of these PTMFs should specify the class name explicitly so that modern compilers can compile the code
```suggestion
        &Act_c::remove_barrel_smoke,
```

Reply:
Updated all Proc arrays to use fully qualified names with 'Act_c::'

Reply:
For future reference, you can just click the "resolve conversation" button to close out a finished comment if you want. Not necessary to type out a response to each one separately.